tests/actions/ReportTest.js

```diff
@@ -0,0 +1,102 @@
+        // channel already in a subscribed state. These methods are normally used to prevent
+        // duplicated subscriptions, but we don't need them for this test.
+        Pusher.isSubscribed = jest.fn().mockReturnValue(false);
+        Pusher.isAlreadySubscribing = jest.fn().mockReturnValue(false);
+
+        // Connect to Pusher
+        PusherConnectionManager.init();
+        Pusher.init({
+            appKey: CONFIG.PUSHER.APP_KEY,
+            cluster: CONFIG.PUSHER.CLUSTER,
+            authEndpoint: `${CONFIG.EXPENSIFY.URL_API_ROOT}api?command=Push_Authenticate`,
+        });
+
+        let reportActions;
+        Onyx.connect({
+            key: `${ONYXKEYS.COLLECTION.REPORT_ACTIONS}${REPORT_ID}`,
+            callback: val => reportActions = val,
+        });
+
+        // Set up Onyx with some test user data
+        return signInWithTestUser(TEST_USER_ACCOUNT_ID, TEST_USER_LOGIN)
+            .then(() => {
+                subscribeToReportCommentEvents();
+                return waitForPromisesToResolve();
+            })
+            .then(() => fetchPersonalDetailsForTestUser(TEST_USER_ACCOUNT_ID, TEST_USER_LOGIN, {
+                [TEST_USER_LOGIN]: {
+                    accountID: TEST_USER_ACCOUNT_ID,
+                    email: TEST_USER_LOGIN,
+                    firstName: 'Test',
+                    lastName: 'User',
+                },
+            }))
+            .then(() => {
+                // This is a fire and forget response, but one it completes we should be able to verify that we
+                // have an "optimistic" report action in Onyx.
+                addAction(REPORT_ID, 'Testing a comment');
+                return waitForPromisesToResolve();
+            })
+            .then(() => {
+                const resultAction = reportActions[ACTION_ID];
+                expect(resultAction).toEqual(REPORT_ACTION);
+            })
+            .then(() => {
+                // Now that we are subscribed we need to simulate a reportComment action Pusher event.
+                // Then verify that action was handled correctly and merged into the reportActions.
+                const channel = Pusher.getChannel('private-user-accountID-1');
+                channel.emit('reportComment', {
+                    reportID: REPORT_ID,
+                    reportAction: REPORT_ACTION,
+                });
+
+                // Once this happens we should see the comment get processed by the callback and added to the
+                // storage so we must wait for promises to resolve again and then verify the data is in Onyx.
+                return waitForPromisesToResolve();
+            })
+            .then(() => {
+                const resultAction = reportActions[ACTION_ID];
+
+                // Verify that our action is no longer in the loading state
+                REPORT_ACTION.loading = false;
+                expect(resultAction).toEqual(REPORT_ACTION);
```

Comment:
Using a different assertion might make it a little more clear what's being tested and prevents the need to modify the contents of `REPORT_ACTION` (which is not quite intuitive as to why it's necessary to modify `REPORT_ACTION`).

Suggestion:
```
expect(resultAction.loading).toEqual(false);
```

Reply:
Yeah this is much better